README.md

```diff
@@ -1,24 +1,52 @@
+
 # IPsec-Recipe
+
 The IPsec recipe is an application that enables strongSwan to use [Infrastructure Application Interface](https://ipdk.io/documentation/Interfaces/InfraApp/), specifically the P4Runtime and OpenConfig gRPCs provided by the [Networking-Recipe](https://github.com/ipdk-io/networking-recipe).
 
 The strongSwan plugin available in this repository implements a policy-based IPsec enablement using security policy database (SPD) and security association database (SAD). The P4Runtime client introduces a programmable IPsec flow using P4 language for the SPD and OpenConfig for SAD configurability.
 
 * IPsec recipe is validated only on Intel&reg; IPU E2100 target
 * Refer to https://ipdk.io/documentation/Recipes/InlineIPsec/
-* YANG model for IPsec SAD: https://github.com/ipdk-io/openconfig-public/blob/master/release/models/ipsec/openconfig-ipsec-offload.yang
+* YANG model for IPsec offload: https://github.com/ipdk-io/openconfig-public/blob/master/release/models/ipsec/openconfig-ipsec-offload.yang
 * Reference P4 program to enable IPsec on DPDK target: https://github.com/ipdk-io/networking-recipe/tree/main/p4src/Inline_IPsec
 
-strongSwan Integration with Inline Crypto Engine
-========================================================
+## strongSwan Integration with Inline Crypto Engine
+
 * The user should start and stop the strongSwan services as described in strongSwan manual.
-* The strongSwan configuration files must be configured by user as described in 
+* The strongSwan configuration files must be configured by user as described in
 			https://wiki.strongswan.org/projects/strongswan/wiki/UserDocumentation
 * Confirm that the networking-recipe (the `infrap4d` process, which acts as the gRPC server) is running before starting strongSwan services.
-* strongSwan hosts (Host-A<---->Host-B) should be connected to negotiate the keys.
-* The user should follow best practices followed by opensource community for other control plane related configurations.
- 
-Getting Started
-===================
+* strongSwan hosts (Host-A<---->Host-B) must be connected to negotiate the keys.
+* The user must follow best practices followed by opensource community for other control plane related configurations.
+
+## Getting Started
+
+### Prerequisites
+
+It is assumed that the ES2K IPU SDE dependencies have been installed on the
+system. The list of dependencies can be found
+in `sde/tools/setup/install_dep.py` file.
+
```

Comment:
is this file part of opensource Networking recipe ? if not then we should not provide mev specific path here.

Reply:
We can also take an alternative approach here -- similar to what the SDE team does.

All the dependencies are listed inside a Python file and the user only needs to run the Python script to run and check to install all dependencies. This is a good approach, since it also makes individual component more "complete". Plus it is idempotent -- if those libs/packages are already installed, it will skip over as nothing to do.

If you are ok with this idea, I can pull those scripts in from SDE repo and check-in here to do the job instead of having the user run multiple steps manually.